src/main/java/com/templateproject/api/controller/CategoryController.java

```diff
@@ -0,0 +1,27 @@
+package com.templateproject.api.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import com.templateproject.api.entity.Category;
+
+import com.templateproject.api.repository.CategoryRepository;
+
+@RestController
+public class CategoryController {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @GetMapping("/categories")
+    public List<Category> index() {
+        return (List<Category>) categoryRepository.findAll();
```

Comment:
Since the return type of the getAllCategories() method is declared as List, there is no need for explicit casting.